lib/repl.js

```diff
@@ -585,7 +585,15 @@ REPLServer.prototype.createContext = function() {
     context = global;
   } else {
     context = vm.createContext();
-    for (var i in global) context[i] = global[i];
+    // #7353 - tab complete returns fewer results
+    context.global = global;
+    vm.runInContext(`
+      (function() {
+        for (let name of Object.getOwnPropertyNames(global)) {
+          this[name] = this[name];
```

Comment:
Is this used to make the properties of `global` enumerable? If so, that’s probably an unwanted side effect here (if not… could you explain what this does?).